legal-api/src/legal_api/services/filings/validations/incorporation_application.py

```diff
@@ -227,6 +237,48 @@ def validate_incorporation_effective_date(incorporation_json) -> Error:
     return None
 
 
+def validate_cooperative_documents(incorporation_json):
+    """Return an error or warning message based on the cooperative documents validation rules.
+
+    Rules:
+        - The documents are provided.
+        - Document IDs are unique.
+    """
+    # Setup
+    msg = []
+
+    rules_file_key = incorporation_json['filing']['incorporationApplication']['cooperative']['rulesFileKey']
+    rules_file_name = incorporation_json['filing']['incorporationApplication']['cooperative']['rulesFileName']
+    memorandum_file_key = incorporation_json['filing']['incorporationApplication']['cooperative']['memorandumFileKey']
+    memorandum_file_name = incorporation_json['filing']['incorporationApplication']['cooperative']['memorandumFileName']
+
+    # Validate key values exist
+    if not rules_file_key:
+        msg.append({'error': babel('A valid rules key is required.')})
+
+    if not rules_file_name:
+        msg.append({'error': babel('A valid rules file name is required.')})
+
+    if not memorandum_file_key:
+        msg.append({'error': babel('A valid memorandum key is required.')})
+
+    if not memorandum_file_name:
+        msg.append({'error': babel('A valid memorandum file name is required.')})
+
+    if msg:
+        return msg
+
+    rules_err = validate_pdf(rules_file_key)
+    if rules_err:
+        return rules_err
+
+    memorandum_err = validate_pdf(memorandum_file_key)
+    if memorandum_err:
+        return memorandum_err
```

Comment:
potentially update so you can return all validation errors in one go?  maybe keep have a `rules_files_exists` and `memorandum_file_exists` flag to decide whether to call the `validate_pdf` functions.

Reply:
Absolutely, will try!

Reply:
So i'm unsure about this one. I do like the accurate error messages without them adding up.

If we fail to have a key value we return with the msg and don't validate the form in the scenario of a bad key. 
From the refactoring i've tried we end up with more conditions if we want to accumulate everything before returning.  
Are you ok with how it currently is?

Reply:
yeah that's fine.